atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1851,13 +1851,12 @@ private void internalPutMetadataForTables(
         final Map<Cell, byte[]> newMetadata = Maps.newHashMap();
         final Collection<CfDef> updatedCfs = Lists.newArrayList();
         for (Entry<Cell, byte[]> entry : metadataRequestedForUpdate.entrySet()) {
-            Value val = persistedMetadata.get(entry.getKey());
-            if (val == null || !Arrays.equals(val.getContents(), entry.getValue())) {
+            if (newMetadataFound(persistedMetadata.get(entry.getKey()), entry.getValue())) {
                 newMetadata.put(entry.getKey(), entry.getValue());
-                updatedCfs.add(getCfForTable(
-                        tableReferenceFromBytes(entry.getKey().getRowName()),
-                        entry.getValue()));
             }
+            updatedCfs.add(getCfForTable(
+                    tableReferenceFromBytes(entry.getKey().getRowName()),
+                    entry.getValue()));
```

Comment:
I'm a little worried moving this out of the block above will lead to schema disagreement issues when concurrently starting up multiple Atlas clients, and Cassandra will be in an unhappy state requiring manual intervention.